app/scripts/controllers/metametrics.js

```diff
@@ -144,16 +147,54 @@ export default class MetaMetricsController {
     // within the fragment's timeout window. When creating a new event fragment
     // a timeout can be specified that will cause an abandoned event to be
     // tracked if the event isn't progressed within that amount of time.
-    setInterval(() => {
-      Object.values(this.store.getState().fragments).forEach((fragment) => {
-        if (
-          fragment.timeout &&
-          Date.now() - fragment.lastUpdated / 1000 > fragment.timeout
-        ) {
-          this.finalizeEventFragment(fragment.id, { abandoned: true });
+    if (isManifestV3) {
+      /* eslint-disable no-undef */
+      chrome.alarms.getAll((alarms) => {
+        const hasAlarm = checkAlarmExists(
+          alarms,
+          METAMETRICS_FINALIZE_EVENT_FRAGMENT_ALARM,
+        );
+
+        if (!hasAlarm) {
+          chrome.alarms.create(METAMETRICS_FINALIZE_EVENT_FRAGMENT_ALARM, {
+            delayInMinutes: 1,
+            periodInMinutes: 1,
+          });
         }
       });
-    }, SECOND * 30);
+      chrome.alarms.onAlarm.addListener(() => {
+        chrome.alarms.getAll((alarms) => {
+          const hasAlarm = checkAlarmExists(
+            alarms,
+            METAMETRICS_FINALIZE_EVENT_FRAGMENT_ALARM,
+          );
+
+          if (hasAlarm) {
+            Object.values(this.store.getState().fragments).forEach(
```

Comment:
Can we make this `Object.values.forEach` block a helper function?  It looks duplicated here and within the `setTimeout`

Reply:
@NiranjanaBinoy we could make a 'finalizeAbandonedFragments' method that does this.